packages/flutter/lib/src/material/circle_avatar.dart

```diff
@@ -142,6 +155,17 @@ class CircleAvatar extends StatelessWidget {
     return 2.0 * (radius ?? maxRadius ?? _defaultMaxRadius);
   }
 
+  BoxBorder get _border {
+    if (borderColor == Colors.transparent || borderWidth == 0.0) {
```

Comment:
I think this should be `&&` since if  borderWidth is non-zero one could reasonably assume that space would be allocated for the border, not matter what its color is

Reply:
With the change to a nullable borderWidth and your point about a transparent border, that check should probably reduce to simply asking if borderWidth is null, right?

Reply:
Yes